Вторая лабораторная

MAYER-Ti opened the pull request:
Реализация аннотации @Repeat и механизма вызова аннотированных методов

MAYER-Ti committed:
Вторая лабораторная

AlekseiMaslakov commented on lab_2/src/lab_2/Main.java:
```diff
@@ -0,0 +1,69 @@
+import java.lang.reflect.*;
+
+public class Main {
+    public static void main(String[] args) throws Exception {
```
Исключения пробрасываются на уровень выше, чтобы их обработать в месте вызова, вместо непосредственно метода, а как обработать то что кидается из main-метода?

MAYER-Ti committed:
Обработка исключений в main

- убран throws Exception в main
- добавлен try-catch с обработкой ClassNotFoundException, NoSuchMethodException, InvocationTargetException
- теперь ошибки не падают в JVM, а выводятся пользователю
- добавлен тестовый метод в DemoService для генерации исключения
